src/main/java/com/conveyal/r5/streets/LinkedPointSet.java

```diff
@@ -491,31 +508,40 @@ public void makeStopToPointLinkageCostTables(Geometry treeRebuildZone) {
                 return stopToPointLinkageCostTables.get(stopIndex);
             }
 
+            counter.increment();
             Envelope distanceTableZone = stopPoint.getEnvelopeInternal();
             GeometryUtils.expandEnvelopeFixed(distanceTableZone, linkingDistanceLimitMeters);
 
-            int[] linkageCostToPoints;
-
             if (streetMode == StreetMode.WALK) {
                 // Walking distances from stops to street vertices are saved in the transitLayer.
                 // Get the pre-computed distance table from the stop to the street vertices,
                 // then extend that table out from the street vertices to the points in this PointSet.
+                // TODO reuse the code that computes the walk tables at com.conveyal.r5.transit.TransitLayer.buildOneDistanceTable() rather than duplicating it below for other modes.
                 TIntIntMap distanceTableToVertices = transitLayer.stopToVertexDistanceTables.get(stopIndex);
-                linkageCostToPoints = distanceTableToVertices == null ? null :
+                return distanceTableToVertices == null ? null :
                         extendDistanceTableToPoints(distanceTableToVertices, distanceTableZone);
-
             } else {
 
                 StreetRouter sr = new StreetRouter(transitLayer.parentNetwork.streetLayer);
                 sr.streetMode = streetMode;
-                sr.setOrigin(transitLayer.streetVertexForStop.get(stopIndex));
+                int vertexId = transitLayer.streetVertexForStop.get(stopIndex);
+                if (vertexId < 0) {
+                    LOG.warn("Stop unlinked, cannot build distance table: {}", stopIndex);
+                    return null;
+                }
+                // TODO setting the origin point of the router to the stop vertex does not work.
+                // This is probably because link edges do not allow car traversal. We could traverse them.
```

Comment:
Would link edges need to consider permissions of the street edges they connect to?  See comment at https://github.com/conveyal/r5/blob/v4.5.3/src/main/java/com/conveyal/r5/streets/Split.java#L101